lib/blockchain.go

```diff
@@ -482,6 +482,13 @@ type Blockchain struct {
 	bestHeaderChain    []*BlockNode
 	bestHeaderChainMap map[BlockHash]*BlockNode
 
+	// Track the highest QC we've seen.
+	highestQC *QuorumCertificate
```

Comment:
Do we need to store this separately? Alternatively, you could dynamically return by having a getter function that returns: `bestChain[len(bestChain) - 1].Header.GetQC()`.

If we can avoid storing it as a duplicate here, it would be ideal. But NBD either way

Reply:
Yeah this makes sense. We don't want our highestQC to be from a possible fork, so just using the best chain is sufficient.